docs/synapse/deploymentguide.rst

```diff
@@ -0,0 +1,363 @@
+
+.. note::
+    It is important that you choose a name and stick with it for a given deployment. Once we begin generating
+    host and service account certificates, changing this name will be difficult.
+
+Deploy AHA Service
+==================
+
+The AHA service is used for service discovery and acts as a CA to issue host/user certificates used to link
+Synapse services. Other Synapse services will need to be able to resolve the IP address of the AHA service
+by name, so it is likely that you need to create a DNS A/AAAA record in your existing resolver. When you are
+using AHA, the only host that needs DNS or other external name resolution is the AHA service.
+
+For this example deployment, we will name our AHA server ``aha.loop.vertex.link`` and assume DNS records
+have been created to resolve the FQDN to an IPv4 / IPv6 address of the host running the container.
+
+Create the container directory::
+
+    mkdir -p /srv/syn/aha/storage
+
+Create the ``/srv/syn/aha/docker-compose.yaml`` file with contents::
+
+    version: "3.3"
+    services:
+      aha:
+        user: "999"
+        image: vertexproject/synapse-aha:v2.x.x
+        network_mode: host
+        restart: unless-stopped
+        volumes:
+            - ./storage:/vertex/storage
+        environment:
+            - SYN_AHACELL_HTTPS_PORT=null
+            - SYN_AHACELL_AHA_NAME=aha
+            - SYN_AHACELL_AHA_network=loop.vertex.link
+            - SYN_AHACELL_DMON_LISTEN=ssl://0.0.0.0?hostname=aha.loop.vertex.link&ca=loop.vertex.link
+            - SYN_AHACELL_PROVISION_LISTEN=ssl://0.0.0.0:27272?hostname=aha.loop.vertex.link
+
+.. note::
+
+    Don't forget to replace ``loop.vertex.link`` with your chosen network name!
+
+Change ownership of the storage directory to the user you will use to run the container::
+
+    chown -R 999 /srv/syn/aha/storage
+
+Start the container using ``docker-compose``::
+
+    docker-compose -f /srv/syn/aha/docker-compose.yaml pull
+    docker-compose -f /srv/syn/aha/docker-compose.yaml up -d
+
+To view the container logs at any time you may run the following command on the *host* from the
+``/srv/syn/aha`` directory::
+
+    docker-compose logs -f
+
+You may also execute a shell inside the container using ``docker-compose`` from the ``/srv/syn/aha``
+directory on the *host*. This will be necessary for some of the additional provisioning steps::
+
+    docker-compose exec aha /bin/bash
+
+Deploy Axon Service
+===================
+
+In the Synapse service architecture, an Axon provides a place to store arbitrary bytes/files as binary
+blobs and exposes APIs for streaming files in and out regardless of their size.  Given sufficient file system
+size, an Axon can be used to efficiently store and retrieve very large files as well as a high number
+(easily billions) of files.
+
+**Inside the AHA container**
+
+Generate a one-time use provisioning URL::
+
+    python -m synapse.tools.aha.provision.service 00.axon
```

Comment:
We can also show this directly:

```
docker exec -it aha-aha-1 python3 -m synapse.tools.aha.provision.service 00.axon
one-time use URL: ssl://aha.yourdomain.com:27272/3b85d15756181327a5d5711c6fd54ef6?certhash=e3ab2e5c83e5e5074eb83f6d63da7c3a275ad94a0f4c2652daa7cec9463b6583
```

Reply:
Trying to keep commands short and decomposed :)